code/modules/ruins/mystery_box.dm

```diff
@@ -0,0 +1,293 @@
+		if(92)
+			new /obj/item/weapon/katana(src)
+		if(93)
+			new /obj/item/weapon/dnainjector/xraymut(src)
+		if(94)
+			new /obj/item/weapon/storage/backpack/clown(src)
+			new /obj/item/clothing/under/rank/clown(src)
+			new /obj/item/clothing/shoes/clown_shoes(src)
+			new /obj/item/device/pda/clown(src)
+			new /obj/item/clothing/mask/gas/clown_hat(src)
+			new /obj/item/weapon/bikehorn(src)
+			new /obj/item/toy/crayon/rainbow(src)
+			new /obj/item/weapon/reagent_containers/spray/waterflower(src)
+		if(95)
+			new /obj/item/clothing/under/rank/mime(src)
+			new /obj/item/clothing/shoes/sneakers/black(src)
+			new /obj/item/device/pda/mime(src)
+			new /obj/item/clothing/gloves/color/white(src)
+			new /obj/item/clothing/mask/gas/mime(src)
+			new /obj/item/clothing/head/beret(src)
+			new /obj/item/clothing/suit/suspenders(src)
+			new /obj/item/toy/crayon/mime(src)
+			new /obj/item/weapon/reagent_containers/food/drinks/bottle/bottleofnothing(src)
+		if(96)
+			new /obj/item/weapon/hand_tele(src)
+		if(97)
+			new /obj/item/clothing/mask/balaclava
+			new /obj/item/weapon/gun/projectile/automatic/pistol(src)
+			new /obj/item/ammo_box/magazine/m10mm(src)
+		if(98)
+			new /obj/item/weapon/katana/cursed(src)
+		if(99)
+			new /obj/item/weapon/storage/belt/champion(src)
+			new /obj/item/clothing/mask/luchador(src)
+		if(100)
+			new /obj/item/clothing/head/bearpelt(src)
+
+/obj/item/weapon/storage/briefcase/mystery/attack_hand(mob/user)
+	if(locked)
+		user << "<span class='notice'>You try entering some digits on the keypad.</span>"
+		var/input = input(usr, "Enter [codelen] digits.", "Deca-Code Lock", "") as text
+		if(user.canUseTopic(src, 1))
+			if (input == code)
+				user << "<span class='notice'>The [src] unlocks!</span>"
+				unlock()
+			else
+				src.audible_message("The [src] makes an annoyed buzzing sound.", "A red light flashes on the [src].")
+				playsound(loc, 'sound/machines/buzz-two.ogg', 50, 0)
+				attempts--
+				if(attempts == 0)
+					boom(user)
+	else
+		return ..()
+
+/obj/item/weapon/storage/briefcase/mystery/proc/unlock()
+	src.audible_message("Tinny congratulatory music plays, as you hear the [src] unlock.", "The lights on the [src] all go green.")
+	playsound(loc, 'sound/effects/yourwinner.ogg', 50, 0)
+	locked = FALSE
+	desc += " The lights on it are all green."
+	// It's not relockable, it is now an ordinary briefcase
+
+
+/obj/item/weapon/storage/briefcase/mystery/Destroy()
+	// If for any reason it's blown up without playing by the rules,
+	// you only get the youtried wrapper
+	if(locked)
+		for(var/atom/movable/AM in src)
+			qdel(AM)
+
+		src += new /obj/item/trash/candy/youtried
+	return ..()
+
+/obj/item/weapon/storage/briefcase/mystery/attack_animal(mob/user)
+	if(locked)
+		boom(user)
+	else
+		return..()
+
+/obj/item/weapon/storage/briefcase/mystery/attackby(obj/item/weapon/W, mob/user)
+	if(locked)
+		if(istype(W, /obj/item/weapon/card/emag))
+			// No.
+			boom(user)
```

Comment:
I've seen both use of reacting to emags in the code, is emag_act the "correct" way?